### DISCUSSION TITLE
Control individual leds in a string

### DISCUSSION THREAD

**Discussion by myozone** [arendst/Tasmota#22298] [Q&A]
I'm trying to figure out how to control individual LEDs in a string of 8 (or more) for events in Node-red via MQTT. sort of LEDx 255,0,255 where x is the the LED and RGB

**Comment by sfromis** [comment]
For addressable pixels, your "sort of" example is **literally** how it works, with the [Led](https://tasmota.github.io/docs/Commands/#led) command accepting a color, or a list of color values for successive pixels.

With ESP32, you also get the Berry language, including extra neopixel features.

**Reply by myozone** [reply]
Well, I didn't expect that! So for example in Node-red msg.payload = LED1, 255,255,255 would be LED1 white or msg.payload = LED1 #FFFFFF ?  I'm going to order some pixels to play with.

**Reply by sfromis** [reply]
Well, the usual ways for commands via MQTT also applies here, meaning the command name (including index for pixel number) included as the last level of the topic, and then the command arguments in payload, in the usual format for the `Color` command, which includes both of the formats you mentioned. You *can* also use `Backlog` as the command name in the topic, and then a semicolon-delimited list of full commands in the payload.

**Reply by myozone** [reply]
I think I understand ... Do you have an example of what you mean exactly? All I really need is have the option of LED number and it's RGB value changeable via a flow.

**Reply by sfromis** [reply]
Your own examples are it. And see https://tasmota.github.io/docs/MQTT/#commands-over-mqtt

**Reply by myozone** [reply]
With a bit of help from AI I get it now - I think :-) ...

Thanks for pointing it all out !
